dashboard/src/components/partials/ChecksList/ChecksList.js

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import PropTypes from "prop-types";
+import gql from "graphql-tag";
+
+import Button from "@material-ui/core/Button";
+import DropdownArrow from "@material-ui/icons/ArrowDropDown";
+import RootRef from "@material-ui/core/RootRef";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+import Typography from "@material-ui/core/Typography";
+
+import { TableListEmptyState } from "/components/TableList";
+import ButtonSet from "/components/ButtonSet";
+
+import Loader from "/components/util/Loader";
+
+import IconButton from "/components/base/IconButton";
+
+import MenuController from "/components/controller/MenuController";
+import ListController from "/components/controller/ListController";
+
+import Pagination from "/components/partials/Pagination";
+import ListHeader from "/components/partials/ListHeader";
+import SilenceEntryDialog from "/components/partials/SilenceEntryDialog";
+import ListSortMenu from "/components/partials/ListSortMenu";
+
+import ChecksListItem from "./ChecksListItem";
+
+class ChecksList extends React.Component {
+  static propTypes = {
+    environment: PropTypes.shape({
+      checks: PropTypes.shape({
+        nodes: PropTypes.array.isRequired,
+      }),
+    }),
+    loading: PropTypes.bool,
+    onChangeQuery: PropTypes.func.isRequired,
+    limit: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    offset: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  };
+
+  static defaultProps = {
+    environment: null,
+    loading: false,
+    limit: undefined,
+    offset: undefined,
+  };
+
+  static fragments = {
+    environment: gql`
+      fragment ChecksList_environment on Environment {
+        checks(
+          limit: $limit
+          offset: $offset
```

Comment:
Pagination does not appear to the be working. May be related to the `@connection` directive?

Reply:
I was missing a `fetchPolicy` attribute on the `<Query>` tag. Fixed now.